src/Common/src/Interop/Unix/System.Private.CoreLib.Native/Interop.MemReAlloc.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal unsafe partial class Sys
+    {
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemReAlloc")]
+        internal static extern IntPtr MemReAlloc(IntPtr ptr, UIntPtr newSize);
+        
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemGetAllocatedSize")]
+        internal static extern IntPtr MemGetAllocatedSize(IntPtr ptr);
+    }
+
+    internal unsafe static IntPtr MemReAlloc(IntPtr ptr, UIntPtr newSize)
+    {
+        return MemReAlloc(ptr, newSize , 0);
+    }
+    
+    internal static unsafe IntPtr MemReAlloc(IntPtr ptr, UIntPtr newSize ,uint flags)
+    {
+        int originalSize = (int) Interop.Sys.MemGetAllocatedSize(ptr);
+        
+        IntPtr allocatedMemory = Interop.Sys.MemReAlloc(ptr, newSize);
+        if (allocatedMemory == IntPtr.Zero)
+        {
+            throw new OutOfMemoryException();
+        }
+        
+        if (flags == HEAP_ZERO_MEMORY)
```

Comment:
Is HEAP_ZERO_MEMORY the only flag allowed?Shouldn't it be if( flags & HEAP_ZERO_MEMORY) ?